### DISCUSSION TITLE
Firmware 07.57  images for the 7530AX and 7590AX are pulled by AVM

### DISCUSSION THREAD

**Discussion by fraternl** [Freetz-NG/freetz-ng#1021] [Allgemein] [answered]
I've lost the virtual machine which I used to build a firmware for the 7590AX
I created a new VM and am able to compile different flavours again.

Alas, For the 7590AX and 7530AX I can't build a 7.5x anymore.
It's because the AVM-images are not anymore on the location where Freetz tries to get it.

```
Location: https://avm.de/de/Service/Service-Portale/7530AX/FRITZ.Box_7530_AX-07.57.image [following]
--2024-08-08 08:14:30--  https://avm.de/de/Service/Service-Portale/7530AX/FRITZ.Box_7530_AX-07.57.image
Resolving avm.de... 212.42.244.122, 2001:bf0:244:244::122
Connecting to avm.de|212.42.244.122|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
```

Can this be solved and if not, is there anyone with these 2 images?
I don't have them anymore.

**Comment by aus-sua** [comment, marked as answer]
Use Internet Archive/waybackmachine, for example:
7530ax F-OS [07.57 int.](https://web.archive.org/web/20230912221633/http://download.avm.de/fritzbox/fritzbox-7530-ax/other/fritz.os/)
7590ax F-OS [07.57 int.](https://web.archive.org/web/20230905124149/http://download.avm.de/fritzbox/fritzbox-7590-ax/other/fritz.os/)

7530ax F-OS [07.57 Dt.](https://web.archive.org/web/20230912221907/http://download.avm.de/fritzbox/fritzbox-7530-ax/deutschland/fritz.os/)
7590ax F-OS [07.57 Dt.](https://web.archive.org/web/20230905125541/http://download.avm.de/fritzbox/fritzbox-7590-ax/deutschland/fritz.os/)

And don't forget:
[Donate Internet Archive/waybackmachine!](https://archive.org/donate?origin=iawww-TopNavDonateButton)

Edit:
url

**Comment by fraternl** [comment]
Many thanks....

I downloaded the 2 files on my webserver and placed them in a folder where it would expect this.

I used this in the hope to find where freetz-ng stored its URL's

```
find -type f -exec grep -l 'avm.de/de/Service/Service-Portale' {} \;
./tools/freetz_download

```

I duplicated the line with 1 of the link (I noticed the incremental counter)
```
                                sites[(( num_sites++ ))]=https://www.avm.de/de/Service/Service-Portale/$subpath
                                sites[(( num_sites++ ))]=https://hmvc.eu/avm/$subpath
```
and started the make process
**It worked** for the **7590AX**, but not for the **7530AX**
I get this message:

```


2024-08-08 19:29:14 (7.44 MB/s) - 'dl/fw/FRITZ.Box_7530_AX-07.57-V2.image' saved [1048576/1048576]

Download succeeded - "https://hmvc.eu/avm/7530AX/FRITZ.Box_7530_AX-07.57.image"  ->  saved to folder "dl/fw"
Checksum disabled for dl/fw/FRITZ.Box_7530_AX-07.57-V2.image

STEP 1: UNPACK
source firmware file: dl/fw/FRITZ.Box_7530_AX-07.57-V2.image
checking signature: /home/freetz/freetz-ng/tools/tar-gnu: Unexpected EOF in archive
/home/freetz/freetz-ng/tools/tar-gnu: Error is not recoverable: exiting now
/home/freetz/freetz-ng/tools/tar-gnu: Unexpected EOF in archive
/home/freetz/freetz-ng/tools/tar-gnu: Error is not recoverable: exiting now
/home/freetz/freetz-ng/tools/tar-gnu: Unexpected EOF in archive
/home/freetz/freetz-ng/tools/tar-gnu: Error is not recoverable: exiting now
failed
OpenSSL: Error opening signature file /tmp/freetz-sig-snU/sig
40E72B45697F0000:error:80000002:system library:BIO_new_file:No such file or directory:../crypto/bio/bss_file.c:67:calling fopen(/tmp/freetz-sig-snU/sig, rb)
40E72B45697F0000:error:10000080:BIO routines:BIO_new_file:no such file:../crypto/bio/bss_file.c:75:
ERROR: image could not be verified with public key
FINISHED

```

Freetz-ng is correct here. 
It turned out that **FRITZ.Box_7530_AX-07.57.image** was only 1 Mbyte :-(
I can probably find another file with the wayback machine.

Tomorrow I will continue.

Thanks a lot!!!

**Comment by fraternl** [comment]
I used the link for the German one and it worked _(at least the build process)
I'm convinced these 2 files should be the same.

I also downloaded the "German" 7590AX file and it was bit for bit the same as the International.
So in the end it all worked out.

It will take until next week before I can actually flash the box as I'm now away from them.

Thanks for the input.


**BTW....**
Some may ask why I used the "webserver approach".
I presumed the machine was caching the AVM images, but when I searched for them using this command:
```
find -type f -name \*.image
```
**I was impatient** and broke it off before it would show me the folder I was looking for, namely  **~/freetz-ng/fw/dl/**
I wrongly assumed it was storing it in some other way, so I decided to find out how it gets them from the Internet.

**Reply by fda77** [reply]
yes, all .image files are in dl/fw/ directory

If you get 
> ERROR: image could not be verified with public key

you could disable verify in menuconfig, but then you are not sure the file is unmodified

**Comment by fraternl** [comment]
In the end it all worked out.
1 file was indeed corrupt and only 1 MByte.
The "_German one_" was 40 MB and passed the verification.

**Comment by fraternl** [comment]
Isn't it an idea to host those files somewhere where they can't be pulled by AVM?

**Comment by fda77** [comment]
AVM told a long time ago this is "illegal". 
I think this is bullshit!
Because the .image files contains GPL parts like kernel, busybox, libraries and programms...

How to find out? You could host all files somewhere and write avm a nice email :)

**Comment by fraternl** [comment]
I think it's bullshit too.
They answered me that "they must have had good reasons to pull it".
Yet, the 7.57 for all the other models are still available.

But even if hosting them is supposed to be illegal, I don't care if you add the link into the repo.
Please tell me if you will, so I will know I have to keep them online.

For now it only contains those 2 images.
I don't think there will be any more downloads than 5 a year for these 2 7.57 images.

BTW....
I was able to flash a factory 7530AX with the 7.8 firmware now.
I can test to see if flashing 7.57 now also works on Monday _(it didn't 3 months ago)_

**Comment by fraternl** [comment]
> How to find out? You could host all files somewhere and write avm a nice email :)

I did just that.
Also pointed out that they are the ones at fault for not providing the image to be able to build the software from source.

**Reply by fda77** [reply]
AVM removes most old  (except some "zwischenupdate) .image files!
